### DISCUSSION TITLE
From Ratchet to ReactPHP

### DISCUSSION THREAD

**Discussion by ghost** [reactphp/reactphp#534] [Q&A]
👋 Hi there,

We're thinking along the lines of refactoring the [PHP Chess Server](https://github.com/chesslablab/chess-server) and use ReactPHP instead of Ratchet. 

Is there any good tutorial on how to do so? 

Thanks for the help,

**Comment by SimonFrings** [comment]
Hey @programarivm, good question, but first I'm interested in why you want to switch from Ratchet to ReactPHP. 

Ratchet itself builds on top of ReactPHP, so you're already using ReactPHP under the hood and switching completely means you would loose the WebSocket part of Ratchet. In conclusion, if you want to keep using WebSockets, you would then have to build the whole thing yourself. 

FYI: We're aware of Ratchet's current state and are already working on compatibility with the newest PHP and PHPUnit versions as described in https://github.com/ratchetphp/Ratchet/issues/1025#issuecomment-1614708487.

One solution could also be [Framework X](https://github.com/clue/framework-x) here, as this project also builds on top of ReactPHP and is very easy to work with. The WebSocket documentation for Framework X is still under construction, but @clue explained how to set this up with Framework X in https://github.com/clue/framework-x/discussions/85#discussioncomment-1712132. In the future we also want to have WebSockets build into Framework X, so it's even easier to use.

Hope this helps :+1:

**Comment by ghost** [comment]
Thank you @SimonFrings, we'd use ReactPHP instead of Ratchet because it is an actively maintained repository.

**Reply by SimonFrings** [reply]
This is also an aspect that we want to change. That's why we are currently exploring sponsorship options (similar to [ReactPHP's sponsor](https://github.com/sponsors/reactphp) concept) to give Ratchet more attention and love 👍

**Comment by ghost** [comment]
Finally, the WebSocket chess server will be sticking to Ratchet because it is just working as expected. Also, we're writing a functional test suite that should also work for a TCP socket. 

We're considering two different solutions: A WebSocket server using Ratchet as well as a TCP socket using ReactPHP.

Thank you, and keep it up!

**Comment by ghost** [comment]
See:

- https://github.com/chesslablab/react-chess/issues/563
- https://github.com/chesslablab/chess-server/issues/153
